parastell/parastell.py

```diff
@@ -291,6 +292,112 @@ def export_source_mesh(self, filename='source_mesh', export_dir=''):
+        if wall_s > 1.0:
+            self.radial_build.radial_build['nwl_geom'] = \
+                self.radial_build.radial_build.pop('sol')
+            del self.radial_build.radial_build['plasma']
+        else:
+            self.radial_build.radial_build['nwl_geom'] = \
+                self.radial_build.radial_build.pop('plasma')
+        
+        self.radial_build._set_mat_tag('nwl_geom', mat_tag)
+
+        self.invessel_build = ivb.InVesselBuild(
+            self._vmec_obj,
+            self.radial_build,
+            logger=self._logger,
+            **kwargs
+        )
+
+        self.invessel_build.populate_surfaces()
+        self.invessel_build.calculate_loci()
+        self.invessel_build.generate_components()
+
+    def export_nwl_build(
+        self, step_filename='nwl_geom', dagmc_filename='nwl_geom', legacy_faceting=True, export_dir='', **kwargs
+    ):
+        """Exports DAGMC neutronics H5M file of ParaStell components via
+        Coreform Cubit.
+
+        Arguments:
+            step_filename (str): name of STEP output file, excluding '.step'
+                extension (optional, defaults to 'nwl_geom').
+            dagmc_filename (str): name of DAGMC output file, excluding '.h5m'
+                extension (optional, defaults to 'dagmc').
+            legacy_faceting (bool): choose legacy or native faceting for DAGMC
+                export (optional, defaults to True).
+            export_dir (str): directory to which to export DAGMC output file
+                (optional, defaults to empty string).
+
+        Optional arguments:
+            faceting_tolerance (float): maximum distance a facet may be from
+                surface of CAD representation for DAGMC export (defaults to
+                None). This attribute is used only for the legacy faceting
+                method.
+            length_tolerance (float): maximum length of facet edge for DAGMC
+                export (defaults to None). This attribute is used only for the
+                legacy faceting method.
+            normal_tolerance (float): maximum change in angle between normal
+                vector of adjacent facets (defaults to None). This attribute is
+                used only for the legacy faceting method.
+            anisotropic_ratio (float): controls edge length ratio of elements
+                (defaults to 100.0). This attribute is used only for the native
+                faceting method.
+            deviation_angle (float): controls deviation angle of facet from
+                surface (i.e., lesser deviation angle results in more elements
+                in areas with higher curvature) (defaults to 5.0). This
+                attribute is used only for the native faceting method.
+        """
+        self.invessel_build.Components[step_filename] = \
+            self.invessel_build.Components.pop('nwl_geom')
+        
+        self.invessel_build.export_step(export_dir=export_dir)
+
+        self.export_dagmc(
```

Comment:
Is there a reason we can't/don't use `cad_to_dagmc` here?

Reply:
CAD-to-DAGMC does not store individual surface IDs in the DAGMC model, instead assigning a single ID to all surfaces in a volume. To tally surface crossings appropriately for NWL, we need to be able to isolate the spline surface.